challenges/index.md

```diff
@@ -0,0 +1,3 @@
+# RSE Leader Challenges
```

Comment:
Would links to resources like https://www.ucl.ac.uk/advanced-research-computing/collaborations-and-consultancy/ways-working be useful to describe how different teams operate?